website/server/libs/payments.js

```diff
@@ -109,7 +117,19 @@ api.addSubToGroupUser = async function addSubToGroupUser (member, group) {
   if (member.isSubscribed()) {
     let memberPlan = member.purchased.plan;
     let customerHasCancelledGroupPlan = memberPlan.customerId === this.constants.GROUP_PLAN_CUSTOMER_ID && !member.hasNotCancelled();
-    if (customerIdsToIgnore.indexOf(memberPlan.customerId) !== -1 && !customerHasCancelledGroupPlan) return;
+    let ignorePaymentPlan = paymentMethodsToIgnore.indexOf(memberPlan.paymentMethod) !== -1;
+    let ignoreCustomerId = customerIdsToIgnore.indexOf(memberPlan.customerId) !== -1;
+
+    if (ignorePaymentPlan) {
+      txnEmail(FLAG_REPORT_EMAILS, 'invalid-payment-cancel', [
+        {name: 'USERNAME', content: member.auth.local.username},
+        {name: 'UUID', content: member._id},
+        {name: 'EMAIL', content: member.auth.local.email},
```

Comment:
This could also be auth.facebook.emails[0] or (possibly) auth.facebook.email (I'm not sure if facebook still adds datta like that) or auth.google.emails[0] or the string "NO_EMAIL" (there are accounts where no address exists). 

I use the code below in my own quick-and-dirty scripts for dumping data about users. I'm including it here just to show the possibilities not because I'm saying it's good enough to be used as-is!
```
function deeptest(myobj, s) {
    s = s.split('.')
    var obj = myobj[s.shift()];
    while(obj && s.length) obj = obj[s.shift()];
    return obj;
}
exports.getUserEmail = function(user) {
    var email = deeptest(user, 'auth.local.email')    ||
                deeptest(user, 'auth.facebook.email') ||
                ( deeptest(user, 'auth.facebook.emails') ?
                    user.auth.facebook.emails[0].value   :
                    '')                       ||
                ( deeptest(user, 'auth.google.emails') ?
                    user.auth.google.emails[0].value   :
                    '')                       ||
                '';

    return email || 'NO_EMAIL';
}
```